camel-k-runtime-core/src/main/java/org/apache/camel/k/support/PropertiesSupport.java

```diff
@@ -26,34 +26,84 @@
 import java.nio.file.SimpleFileVisitor;
 import java.nio.file.attribute.BasicFileAttributes;
 import java.util.Collection;
+import java.util.HashMap;
 import java.util.LinkedHashSet;
 import java.util.Map;
 import java.util.Objects;
 import java.util.Properties;
 import java.util.Set;
+import java.util.function.Predicate;
 
 import org.apache.camel.CamelContext;
+import org.apache.camel.Component;
+import org.apache.camel.ExtendedCamelContext;
 import org.apache.camel.k.Constants;
 import org.apache.camel.spi.PropertiesComponent;
+import org.apache.camel.spi.PropertyConfigurer;
 import org.apache.camel.support.PropertyBindingSupport;
+import org.apache.camel.support.service.ServiceHelper;
 import org.apache.camel.util.ObjectHelper;
+import org.apache.camel.util.StringHelper;
 
 public final class PropertiesSupport {
     private PropertiesSupport() {
     }
 
-    @SuppressWarnings("unchecked")
-    public static boolean bindProperties(CamelContext context, Object target, String prefix) {
+    public static <T> T bindProperties(CamelContext context, T target, String prefix) {
+        return bindProperties(context, target, prefix, false);
+    }
+
+    public static <T> T bindProperties(CamelContext context, T target, String prefix, boolean stripPrefix) {
+        return bindProperties(context, target, k -> k.startsWith(prefix), prefix, stripPrefix);
+    }
+
+    public static <T> T bindProperties(CamelContext context, T target, Predicate<String> filter, String prefix) {
+        return bindProperties(context, target, filter, prefix, false);
+    }
+
+    public static <T> T bindProperties(CamelContext context, T target, Predicate<String> filter, String prefix, boolean stripPrefix) {
         final PropertiesComponent component = context.getPropertiesComponent();
-        final Properties properties = component.loadProperties(k -> k.startsWith(prefix));
+        final Properties propertiesWithPrefix = component.loadProperties(filter);
+        final Map<String, Object> properties = new HashMap<>();
+
+        propertiesWithPrefix.stringPropertyNames().forEach(
+            name -> properties.put(
+                // PropertyBindingSupport does not support dash properties
+                StringHelper.dashToCamelCase(stripPrefix ? name.substring(prefix.length()) : name),
+                propertiesWithPrefix.getProperty(name))
+        );
+
+        PropertyConfigurer configurer = null;
+        if (target instanceof Component) {
+            // the component needs to be initialized to have the configurer ready
+            ServiceHelper.initService(target);
+            configurer = ((Component) target).getComponentPropertyConfigurer();
+        }
+
+        if (configurer == null) {
+            String name = target.getClass().getSimpleName();
```

Comment:
Using long name now, see the code in camel-core

Reply:
on it